src/transformers/models/splinter/modeling_splinter.py

```diff
@@ -940,3 +941,160 @@ def forward(
+            are not taken into account for computing the loss.
+        question_positions (`torch.LongTensor` of shape `(batch_size, num_questions)`, *optional*):
+            The positions of all question tokens. If given, start_logits and end_logits will be of shape `(batch_size,
+            num_questions, sequence_length)`. If None, the first question token in each sequence in the batch will be
+            the only one for which start_logits and end_logits are calculated and they will be of shape `(batch_size,
+            sequence_length)`.
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        if question_positions is None:
+            if start_positions is not None and end_positions is not None:
+                raise TypeError(
+                    "For span selection training, question_positions must be specified to obtain the correct loss"
+                )
+
+            if input_ids is not None:
+                question_positions = torch.argmax(
+                    (torch.eq(input_ids, self.question_token_id)).int(), dim=-1
+                ).unsqueeze(0)
+            else:
+                raise TypeError("question_positions is required when input_embeds is specified")
+
+        outputs = self.splinter(
+            input_ids,
+            attention_mask=attention_mask,
+            token_type_ids=token_type_ids,
+            position_ids=position_ids,
+            head_mask=head_mask,
+            inputs_embeds=inputs_embeds,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        sequence_output = outputs[0]
+        batch_size, sequence_length, dim = sequence_output.size()
+        start_logits, end_logits = self.splinter_qass(sequence_output, question_positions)  # [B, Q, S]
+
+        num_questions = question_positions.size(1)
+        if attention_mask is not None:
+            attention_mask_for_each_question = attention_mask.unsqueeze(1).expand(
+                batch_size, num_questions, sequence_length
+            )
+            start_logits = start_logits + (1 - attention_mask_for_each_question) * -10000.0
+            end_logits = end_logits + (1 - attention_mask_for_each_question) * -10000.0
+
+        total_loss = None
+        if start_positions is not None and end_positions is not None:  # [B, Q, S]
+            # sometimes the start/end positions are outside our model inputs, we ignore these terms
+            start_positions.clamp_(0, max(0, sequence_length - 1))
+            end_positions.clamp_(0, max(0, sequence_length - 1))
+
+            # Ignore zero positions in the loss. Splinter never predicts zero
+            # during pretraining and zero is used for padding question
+            # tokens as well as for start and end positions of padded
+            # question tokens.
+            loss_fct = CrossEntropyLoss(ignore_index=0)
```

Comment:
Think we could use `config.pad_token_id` instead of hardcoding 0 here.

Reply:
done